libcrux-ml-dsa/src/simd/portable/encoding/commitment.rs

```diff
@@ -1,40 +1,161 @@
 use crate::{helper::cloop, simd::portable::vector_type::Coefficients};
 
 #[inline(always)]
-pub fn serialize(simd_unit: &Coefficients, serialized: &mut [u8]) {
-    match serialized.len() as u8 {
-        4 => {
-            // The commitment has coefficients in [0,15] => each coefficient occupies
-            // 4 bits.
-            cloop! {
-                for (i, coefficients) in simd_unit.values.chunks_exact(2).enumerate() {
-                    let coefficient0 = coefficients[0] as u8;
-                    let coefficient1 = coefficients[1] as u8;
-
-                    serialized[i] = (coefficient1 << 4) | coefficient0;
-                }
-            }
-            ()
-        }
+#[hax_lib::fstar::before(
+    r#"
+#set-options "--fuel 0 --ifuel 1 --z3rlimit 300 --z3version 4.13.3 --ext context_pruning"
```

Comment:
Do we really depend on a specific version of z3? I would prefer to leave such version setting to the Makefile

Reply:
Would be happy to use a recent Z3 by default (I've set that locally not to mess with anything else)

Reply:
Let's just remove the version here then. System default is fine, and any version pinning should happend outside the Rust files.